### PR TITLE
envoy: Fix data race in RegisterServiceUsageInCEC

### LINKED ISSUE #33900: [DATA RACE]: github.com/cilium/cilium/pkg/ciliumenvoyconfig.(*envoyServiceBackendSyncer).RegisterServiceUsageInCEC()
```
2024-07-18T14:40:02.213010171Z WARNING: DATA RACE
2024-07-18T14:40:02.213013087Z Write at 0x00c006155c50 by goroutine 2895:
2024-07-18T14:40:02.213015581Z   runtime.mapassign()
2024-07-18T14:40:02.213042372Z       /usr/local/go/src/runtime/map.go:579 +0x0
2024-07-18T14:40:02.213045978Z   github.com/cilium/cilium/pkg/ciliumenvoyconfig.(*envoyServiceBackendSyncer).RegisterServiceUsageInCEC()
2024-07-18T14:40:02.213048312Z       /go/src/github.com/cilium/cilium/pkg/ciliumenvoyconfig/envoy_l7lb_backend_syncer.go:89 +0x1f6
2024-07-18T14:40:02.213050877Z   github.com/cilium/cilium/pkg/ciliumenvoyconfig.(*cecManager).registerServiceSync()
2024-07-18T14:40:02.213053031Z       /go/src/github.com/cilium/cilium/pkg/ciliumenvoyconfig/cec_manager.go:479 +0x14a
2024-07-18T14:40:02.213055446Z   github.com/cilium/cilium/pkg/ciliumenvoyconfig.(*cecManager).syncCiliumEnvoyConfigService()
2024-07-18T14:40:02.213057660Z       /go/src/github.com/cilium/cilium/pkg/ciliumenvoyconfig/cec_manager.go:166 +0x27e
2024-07-18T14:40:02.213059774Z   github.com/cilium/cilium/pkg/ciliumenvoyconfig.(*ciliumEnvoyConfigReconciler).syncHeadlessService()
2024-07-18T14:40:02.213061938Z       /go/src/github.com/cilium/cilium/pkg/ciliumenvoyconfig/cec_reconciler.go:306 +0x31e
2024-07-18T14:40:02.213064152Z   github.com/cilium/cilium/pkg/ciliumenvoyconfig.(*ciliumEnvoyConfigReconciler).syncHeadlessService-fm()
2024-07-18T14:40:02.213066567Z       <autogenerated>:1 +0x47
2024-07-18T14:40:02.213068821Z   github.com/cilium/hive/job.(*jobTimer).start()
2024-07-18T14:40:02.213070975Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/job/timer.go:131 +0x6b7
2024-07-18T14:40:02.213073670Z   github.com/cilium/hive/job.(*group).Start.func1.gowrap1()
2024-07-18T14:40:02.213075844Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/job/job.go:159 +0x131
2024-07-18T14:40:02.213077817Z 
2024-07-18T14:40:02.213079981Z Previous read at 0x00c006155c50 by goroutine 710:
2024-07-18T14:40:02.213082316Z   runtime.mapiterinit()
2024-07-18T14:40:02.213084540Z       /usr/local/go/src/runtime/map.go:816 +0x0
2024-07-18T14:40:02.213086774Z   github.com/cilium/cilium/pkg/ciliumenvoyconfig.(*backendSyncInfo).GetAllFrontendPorts()
2024-07-18T14:40:02.213089068Z       /go/src/github.com/cilium/cilium/pkg/ciliumenvoyconfig/envoy_l7lb_backend_syncer.go:242 +0x84
2024-07-18T14:40:02.213131568Z   github.com/cilium/cilium/pkg/ciliumenvoyconfig.(*envoyServiceBackendSyncer).Sync()
2024-07-18T14:40:02.213137649Z       /go/src/github.com/cilium/cilium/pkg/ciliumenvoyconfig/envoy_l7lb_backend_syncer.go:60 +0x148
2024-07-18T14:40:02.213140204Z   github.com/cilium/cilium/pkg/service.(*Service).upsertService()
2024-07-18T14:40:02.213142818Z       /go/src/github.com/cilium/cilium/pkg/service/service.go:825 +0x23f6
2024-07-18T14:40:02.213145033Z   github.com/cilium/cilium/pkg/service.(*Service).UpsertService()
2024-07-18T14:40:02.213147177Z       /go/src/github.com/cilium/cilium/pkg/service/service.go:675 +0xb8
2024-07-18T14:40:02.213149351Z   github.com/cilium/cilium/pkg/k8s/watchers.(*K8sServiceWatcher).addK8sSVCs()
2024-07-18T14:40:02.213151435Z       /go/src/github.com/cilium/cilium/pkg/k8s/watchers/service.go:478 +0x175e
2024-07-18T14:40:02.213153819Z   github.com/cilium/cilium/pkg/k8s/watchers.(*K8sServiceWatcher).k8sServiceHandler.func1()
2024-07-18T14:40:02.213156073Z       /go/src/github.com/cilium/cilium/pkg/k8s/watchers/service.go:184 +0x8ed
2024-07-18T14:40:02.213157977Z   github.com/cilium/cilium/pkg/k8s/watchers.(*K8sServiceWatcher).k8sServiceHandler()
2024-07-18T14:40:02.213159911Z       /go/src/github.com/cilium/cilium/pkg/k8s/watchers/service.go:197 +0xa6
2024-07-18T14:40:02.213161955Z   github.com/cilium/cilium/pkg/k8s/watchers.(*K8sServiceWatcher).RunK8sServiceHandler.gowrap1()
2024-07-18T14:40:02.213163989Z       /go/src/github.com/cilium/cilium/pkg/k8s/watchers/service.go:203 +0x33
2024-07-18T14:40:02.213165712Z 
2024-07-18T14:40:02.213167785Z Goroutine 2895 (running) created at:
2024-07-18T14:40:02.213169920Z   github.com/cilium/hive/job.(*group).Start.func1()
2024-07-18T14:40:02.213172023Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/job/job.go:159 +0x2c4
2024-07-18T14:40:02.213174087Z   runtime/pprof.Do()
2024-07-18T14:40:02.213176291Z       /usr/local/go/src/runtime/pprof/runtime.go:51 +0x117
2024-07-18T14:40:02.213178706Z   github.com/cilium/hive/job.(*group).Start()
2024-07-18T14:40:02.213180870Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/job/job.go:158 +0x353
2024-07-18T14:40:02.213183114Z   github.com/cilium/hive/cell.(*DefaultLifecycle).Start()
2024-07-18T14:40:02.213185518Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/lifecycle.go:107 +0x46a
2024-07-18T14:40:02.213187753Z   fmt.Fscanf()
2024-07-18T14:40:02.213190067Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213192221Z   fmt.Sscanf()
2024-07-18T14:40:02.213194405Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.213196589Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.213198854Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.213200967Z   github.com/cilium/cilium/pkg/bpf.LoadCollectionSpec()
2024-07-18T14:40:02.213203071Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:36 +0x64
2024-07-18T14:40:02.213205225Z   github.com/cilium/cilium/pkg/datapath/loader.(*loader).replaceOverlayDatapath()
2024-07-18T14:40:02.213207239Z       /go/src/github.com/cilium/cilium/pkg/datapath/loader/loader.go:540 +0x305
2024-07-18T14:40:02.213209613Z   github.com/cilium/cilium/pkg/datapath/loader.(*loader).reinitializeOverlay()
2024-07-18T14:40:02.213211848Z       /go/src/github.com/cilium/cilium/pkg/datapath/loader/base.go:304 +0xc10
2024-07-18T14:40:02.213214472Z   github.com/cilium/cilium/pkg/datapath/loader.(*loader).Reinitialize()
2024-07-18T14:40:02.213220173Z       /go/src/github.com/cilium/cilium/pkg/datapath/loader/base.go:520 +0x29b9
2024-07-18T14:40:02.213328916Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213339816Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213418042Z   fmt.Fscanf()
2024-07-18T14:40:02.213421618Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213424053Z   fmt.Sscanf()
2024-07-18T14:40:02.213426638Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213429152Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213431948Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213434132Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213436707Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213439121Z   fmt.Fscanf()
2024-07-18T14:40:02.213441556Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213443900Z   fmt.Sscanf()
2024-07-18T14:40:02.213446315Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213448649Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213450953Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213453288Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213455622Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213457996Z   fmt.Fscanf()
2024-07-18T14:40:02.213460311Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213462756Z   fmt.Sscanf()
2024-07-18T14:40:02.213465220Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213511246Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213514122Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213516636Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213518920Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213521345Z   fmt.Fscanf()
2024-07-18T14:40:02.213523780Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213526274Z   fmt.Sscanf()
2024-07-18T14:40:02.213528528Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213530833Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213533217Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213535421Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213537565Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213539829Z   fmt.Fscanf()
2024-07-18T14:40:02.213542384Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213544768Z   fmt.Sscanf()
2024-07-18T14:40:02.213547163Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213549617Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213551981Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213554346Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213556731Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213559045Z   fmt.Fscanf()
2024-07-18T14:40:02.213561330Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213607586Z   fmt.Sscanf()
2024-07-18T14:40:02.213610662Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213613046Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213615280Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213617855Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213619969Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213622022Z   fmt.Fscanf()
2024-07-18T14:40:02.213624267Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213629236Z   fmt.Sscanf()
2024-07-18T14:40:02.213637552Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213639876Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213642330Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213644714Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213646919Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213649093Z   fmt.Fscanf()
2024-07-18T14:40:02.213651297Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213653542Z   fmt.Sscanf()
2024-07-18T14:40:02.213655836Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213658340Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213706670Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213709375Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213711710Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213714024Z   fmt.Fscanf()
2024-07-18T14:40:02.213716328Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213718803Z   fmt.Sscanf()
2024-07-18T14:40:02.213720917Z       /usr/local/go/src/fmt/scan.go:114 +0x7c4
2024-07-18T14:40:02.213723251Z   github.com/cilium/cilium/pkg/bpf.iproute2Compat()
2024-07-18T14:40:02.213725676Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:220 +0x6a0
2024-07-18T14:40:02.213728060Z   github.com/cilium/cilium/pkg/bpf.LoadCollectionSpec()
2024-07-18T14:40:02.213730655Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:40 +0x77
2024-07-18T14:40:02.213733150Z   fmt.Fscanf()
2024-07-18T14:40:02.213735544Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213737888Z   fmt.Sscanf()
2024-07-18T14:40:02.213740243Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.213742547Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.213744952Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.213747597Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213750101Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213752456Z   fmt.Fscanf()
2024-07-18T14:40:02.213754610Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213756904Z   fmt.Sscanf()
2024-07-18T14:40:02.213801687Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.213805024Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.213807559Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.213809793Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213812177Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213815994Z   fmt.Fscanf()
2024-07-18T14:40:02.213818559Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213820923Z   fmt.Sscanf()
2024-07-18T14:40:02.213823138Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.213825602Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.213827786Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.213830421Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213832956Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213835330Z   fmt.Fscanf()
2024-07-18T14:40:02.213837865Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213840199Z   fmt.Sscanf()
2024-07-18T14:40:02.213842594Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.213910821Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.213920088Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.213922493Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213925067Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213927532Z   fmt.Fscanf()
2024-07-18T14:40:02.213929917Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213932451Z   fmt.Sscanf()
2024-07-18T14:40:02.213934746Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.213937060Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.213939544Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.213941788Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.213944263Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.213946608Z   fmt.Fscanf()
2024-07-18T14:40:02.213948952Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.213951337Z   fmt.Sscanf()
2024-07-18T14:40:02.213953641Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.213956005Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.214013192Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.214015897Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.214018422Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.214020756Z   fmt.Fscanf()
2024-07-18T14:40:02.214023421Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.214025886Z   fmt.Sscanf()
2024-07-18T14:40:02.214028230Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.214030745Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.214032929Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.214035143Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.214037687Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.214039982Z   fmt.Fscanf()
2024-07-18T14:40:02.214042437Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.214044801Z   fmt.Sscanf()
2024-07-18T14:40:02.214047025Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.214049520Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.214051724Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.214054078Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.214056593Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.214058917Z   fmt.Fscanf()
2024-07-18T14:40:02.214061182Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.214063495Z   fmt.Sscanf()
2024-07-18T14:40:02.214090376Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.214093522Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.214095836Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.214098391Z   fmt.(*ss).doScanf()
2024-07-18T14:40:02.214127455Z       /usr/local/go/src/fmt/scan.go:1230 +0x435
2024-07-18T14:40:02.214135139Z   fmt.Fscanf()
2024-07-18T14:40:02.214137935Z       /usr/local/go/src/fmt/scan.go:143 +0xdc
2024-07-18T14:40:02.214140249Z   fmt.Sscanf()
2024-07-18T14:40:02.214142704Z       /usr/local/go/src/fmt/scan.go:114 +0x355
2024-07-18T14:40:02.214145168Z   github.com/cilium/cilium/pkg/bpf.removeUnreachableTailcalls()
2024-07-18T14:40:02.214147723Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:72 +0x22c
2024-07-18T14:40:02.214149977Z   github.com/cilium/cilium/pkg/bpf.LoadCollectionSpec()
2024-07-18T14:40:02.214157701Z       /go/src/github.com/cilium/cilium/pkg/bpf/collection.go:36 +0x64
2024-07-18T14:40:02.214160196Z   github.com/cilium/cilium/pkg/socketlb.Enable()
2024-07-18T14:40:02.214162430Z       /go/src/github.com/cilium/cilium/pkg/socketlb/socketlb.go:64 +0x210
2024-07-18T14:40:02.214165075Z   github.com/cilium/cilium/pkg/datapath/loader.(*loader).Reinitialize()
2024-07-18T14:40:02.214167540Z       /go/src/github.com/cilium/cilium/pkg/datapath/loader/base.go:487 +0x1fad
2024-07-18T14:40:02.214172058Z   github.com/cilium/cilium/pkg/datapath/orchestrator.(*orchestrator).Reinitialize()
2024-07-18T14:40:02.214174422Z       /go/src/github.com/cilium/cilium/pkg/datapath/orchestrator/orchestrator.go:77 +0x857
2024-07-18T14:40:02.214176837Z   github.com/cilium/cilium/daemon/cmd.(*Daemon).init()
2024-07-18T14:40:02.214179412Z       /go/src/github.com/cilium/cilium/daemon/cmd/daemon.go:211 +0x870
2024-07-18T14:40:02.214181876Z   github.com/cilium/cilium/daemon/cmd.newDaemon()
2024-07-18T14:40:02.214184351Z       /go/src/github.com/cilium/cilium/daemon/cmd/daemon.go:848 +0x8264
2024-07-18T14:40:02.214186866Z   github.com/cilium/cilium/daemon/cmd.newDaemonPromise.func1()
2024-07-18T14:40:02.214189200Z       /go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:1740 +0x1e9
2024-07-18T14:40:02.214191835Z   github.com/cilium/hive/cell.Hook.Start()
2024-07-18T14:40:02.214194019Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/lifecycle.go:42 +0x70
2024-07-18T14:40:02.214196524Z   github.com/cilium/hive/cell.(*Hook).Start()
2024-07-18T14:40:02.214199028Z       <autogenerated>:1 +0x1f
2024-07-18T14:40:02.214201343Z   github.com/cilium/hive/cell.(*DefaultLifecycle).Start()
2024-07-18T14:40:02.214204058Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/lifecycle.go:107 +0x46a
2024-07-18T14:40:02.214206332Z   github.com/cilium/hive.(*Hive).Start()
2024-07-18T14:40:02.214208827Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:339 +0x192
2024-07-18T14:40:02.214211121Z   github.com/cilium/hive.(*Hive).Run()
2024-07-18T14:40:02.214213556Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:229 +0xc9
2024-07-18T14:40:02.214215980Z   github.com/cilium/cilium/daemon/cmd.NewAgentCmd.func1()
2024-07-18T14:40:02.214218284Z       /go/src/github.com/cilium/cilium/daemon/cmd/root.go:40 +0x28f
2024-07-18T14:40:02.214220719Z   github.com/spf13/cobra.(*Command).execute()
2024-07-18T14:40:02.214223124Z       /go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:989 +0x1185
2024-07-18T14:40:02.214225418Z   github.com/spf13/cobra.(*Command).ExecuteC()
2024-07-18T14:40:02.214227702Z       /go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1117 +0x657
2024-07-18T14:40:02.214238993Z   github.com/spf13/cobra.(*Command).Execute()
2024-07-18T14:40:02.214241678Z       /go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1041 +0x2e
2024-07-18T14:40:02.214244283Z   github.com/cilium/cilium/daemon/cmd.Execute()
2024-07-18T14:40:02.214246577Z       /go/src/github.com/cilium/cilium/daemon/cmd/root.go:80 +0x12
2024-07-18T14:40:02.214248942Z   main.main()
2024-07-18T14:40:02.214251416Z       /go/src/github.com/cilium/cilium/daemon/main.go:14 +0xa9
2024-07-18T14:40:02.214253450Z 
2024-07-18T14:40:02.214255894Z Goroutine 710 (running) created at:
2024-07-18T14:40:02.214258359Z   github.com/cilium/cilium/pkg/k8s/watchers.(*K8sServiceWatcher).RunK8sServiceHandler()
2024-07-18T14:40:02.214260724Z       /go/src/github.com/cilium/cilium/pkg/k8s/watchers/service.go:203 +0x8d
2024-07-18T14:40:02.214263319Z   github.com/cilium/cilium/pkg/k8s/watchers.(*K8sWatcher).RunK8sServiceHandler()
2024-07-18T14:40:02.214269960Z       /go/src/github.com/cilium/cilium/pkg/k8s/watchers/watcher.go:390 +0x2724
2024-07-18T14:40:02.214272365Z   github.com/cilium/cilium/daemon/cmd.newDaemon()
2024-07-18T14:40:02.214275000Z       /go/src/github.com/cilium/cilium/daemon/cmd/daemon.go:479 +0x26cc
2024-07-18T14:40:02.214277414Z   github.com/cilium/cilium/daemon/cmd.newDaemonPromise.func1()
2024-07-18T14:40:02.214279689Z       /go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:1740 +0x1e9
2024-07-18T14:40:02.214281994Z   github.com/cilium/hive/cell.Hook.Start()
2024-07-18T14:40:02.214284187Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/lifecycle.go:42 +0x70
2024-07-18T14:40:02.214287103Z   github.com/cilium/hive/cell.(*Hook).Start()
2024-07-18T14:40:02.214289757Z       <autogenerated>:1 +0x1f
2024-07-18T14:40:02.214292072Z   github.com/cilium/hive/cell.(*DefaultLifecycle).Start()
2024-07-18T14:40:02.214294366Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/lifecycle.go:107 +0x46a
2024-07-18T14:40:02.214296861Z   github.com/cilium/hive.(*Hive).Start()
2024-07-18T14:40:02.214299446Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:339 +0x192
2024-07-18T14:40:02.214630423Z   github.com/cilium/hive.(*Hive).Run()
2024-07-18T14:40:02.214634701Z       /go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/hive.go:229 +0xc9
2024-07-18T14:40:02.214637245Z   github.com/cilium/cilium/daemon/cmd.NewAgentCmd.func1()
2024-07-18T14:40:02.214639951Z       /go/src/github.com/cilium/cilium/daemon/cmd/root.go:40 +0x28f
2024-07-18T14:40:02.214642395Z   github.com/spf13/cobra.(*Command).execute()
2024-07-18T14:40:02.214645100Z       /go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:989 +0x1185
2024-07-18T14:40:02.214647665Z   github.com/spf13/cobra.(*Command).ExecuteC()
2024-07-18T14:40:02.214650069Z       /go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1117 +0x657
2024-07-18T14:40:02.214652514Z   github.com/spf13/cobra.(*Command).Execute()
2024-07-18T14:40:02.214654918Z       /go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:1041 +0x2e
2024-07-18T14:40:02.214657233Z   github.com/cilium/cilium/daemon/cmd.Execute()
2024-07-18T14:40:02.214659778Z       /go/src/github.com/cilium/cilium/daemon/cmd/root.go:80 +0x12
2024-07-18T14:40:02.214662152Z   main.main()
2024-07-18T14:40:02.214664587Z       /go/src/github.com/cilium/cilium/daemon/main.go:14 +0xa9
```
seen with https://github.com/cilium/cilium/commit/54796b085eb2f8c695db74cef95521e732effacf

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Fixes: #33900

```release-note
envoy: Fix data race in RegisterServiceUsageInCEC
```

### CODE REVIEW

File: pkg/ciliumenvoyconfig/envoy_l7lb_backend_syncer.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func newEnvoyServiceBackendSyncer(logger logrus.FieldLogger, envoyXdsServer envo
 func (r *envoyServiceBackendSyncer) Sync(svc *loadbalancer.SVC) error {
 	r.l7lbSvcsMutex.RLock()
 	l7lbInfo, exists := r.l7lbSvcs[svc.Name]
+	frontendPorts := l7lbInfo.GetAllFrontendPorts()
```
Comment (sayboras) [inline pkg/ciliumenvoyconfig/envoy_l7lb_backend_syncer.go]: l7lbInfo can be nil and cause panic here.

**Review by sayboras — COMMENTED** [review]
Thanks for your time investigating this issue, I have one comment as per below.

**Review by sayboras — APPROVED** [review]
Thanks and lgtm :heavy_check_mark:

**Comment by joamaki** [comment]
@sayboras v1.15 does not have this file or method. Removed the backport label.

**Comment by sayboras** [comment]
> v1.15 does not have this file or method. Removed the backport label.

Noted and thanks, the logic was in different file, I will do the backport manually.

**Comment by sayboras** [comment]
Seems like we don't need to backport this patch into pkg/service/service.go for 1.15.

**Pull request merged**